File: src/main/java/me/newyith/fortress/generator/rune/GeneratorState.java
package me.newyith.fortress.generator.rune;

import me.newyith.fortressold.util.Debug;

public enum GeneratorState {
	NULL,
	RUNNING,
	PAUSED,
	NEEDS_FUEL;

	//-----------------------------------------------------------------------

	public static int toInt(GeneratorState state) {
		return state.ordinal();
	}

	public static GeneratorState fromInt(int ordinal) {
		GeneratorState fgState = GeneratorState.NULL;

		if (GeneratorState.RUNNING.ordinal() == ordinal) {
			fgState = GeneratorState.RUNNING;
		} else if (GeneratorState.PAUSED.ordinal() == ordinal) {
			fgState = GeneratorState.PAUSED;
		} else if (GeneratorState.NEEDS_FUEL.ordinal() == ordinal) {
			fgState = GeneratorState.NEEDS_FUEL;
		} else {
			Debug.error("ERROR: FgState.fromInt(" + ordinal + ") did not match any state.");
		}

		return fgState;
	}
}

File: src_old/main/java/me/newyith/fortress_try1/rune/generator/GeneratorState.kt
package me.newyith.fortress_try1.rune.generator

enum class GeneratorState {
	NULL,
	RUNNING,
	PAUSED,
	NEEDS_FUEL;

	//TODO: consider uncommenting out (and maybe making back into companion object) toInt() and fromInt() if needed for save/load
//	fun toInt(): Int {
//		return this.ordinal
//	}
//
//	fun fromInt(): GeneratorState {
//		return when (this.ordinal) {
//			GeneratorState.RUNNING.ordinal -> GeneratorState.RUNNING
//			GeneratorState.PAUSED.ordinal -> GeneratorState.PAUSED
//			GeneratorState.NEEDS_FUEL.ordinal -> GeneratorState.NEEDS_FUEL
//			else -> {
//				Log.error("ERROR: state.fromInt(${this.ordinal}) did not match any state.")
//				GeneratorState.NULL
//			}
//		}
//	}
}